Gemfile

```diff
@@ -6,7 +6,9 @@ gem 'danger', '~> 8.0'
 gem 'danger-rubocop', '~> 0.6'
 
 gem 'codecov', require: false
+gem 'gettext', '~> 3.4.3'
```

Comment:
Usually the best practice to specify versions is to use `~> x.y` if possible, to respect semantic versioning (semver.org)

This is because `~> major.minor.patch` is an operator that means "version `major.minor.patch` minimum, you can also get a version with a higher value of the latest digit specified (`patch` in this case), but the other digits (non-last-ones) should not be bumped.

But, given the rules of [Semantic Versioning](semver.org), while bumping the patch or minor version of a lib means it is still backwards compatible, while bumping the major version of a lib means its API got breaking changes hat won't be compatible with the client's call sites, then we want to allow the minor and patch to be higher (so we can get latest fixes for free without risking breaking things) but don't want to allow a major version bump (as it would break the call sites)

So for example `~> 3.4.3` allows version `3.4.3`, and also an hypothetical version `3.4.4` if such one comes out later… but if a new version `3.5.0` of this gem is released later, that one won't be picked, because `3.5.0` increases the minor version number which is not the last version number specified by `~> 3.4.3`.

But if you instead just use `~> 3.4`, then today it will pick `3.4.3` (since that's the latest version of the `gettext` gem at the time I'm writing this comment that matches "`3.x` with `x≥4`"), but in the future it will also allow to pick `3.4.4`, `3.5.0`, `3.5.8`, `3.9.2`… as long as it's `3.x` with `x≥4` (so it won't pick an hypothetical future `4.0.0`, which is a good thing because such a major version bump would mean a breaking change)

Reply:
> Usually the best practice to specify versions is to use `~> x.y` if possible, to respect semantic versioning (semver.org)
> 
> This is because `~> major.minor.patch` is an operator that means "version `major.minor.patch` minimum, you can also get a version with a higher value of the latest digit specified (`patch` in this case), but the other digits (non-last-ones) should not be bumped.
> 
> But, given the rules of [Semantic Versioning](semver.org), while bumping the patch or minor version of a lib means it is still backwards compatible, while bumping the major version of a lib means its API got breaking changes hat won't be compatible with the client's call sites, then we want to allow the minor and patch to be higher (so we can get latest fixes for free without risking breaking things) but don't want to allow a major version bump (as it would break the call sites)
> 
> So for example `~> 3.4.3` allows version `3.4.3`, and also an hypothetical version `3.4.4` if such one comes out later… but if a new version `3.5.0` of this gem is released later, that one won't be picked, because `3.5.0` increases the minor version number which is not the last version number specified by `~> 3.4.3`.
> 
> But if you instead just use `~> 3.4`, then today it will pick `3.4.3` (since that's the latest version of the `gettext` gem at the time I'm writing this comment that matches "`3.x` with `x≥4`"), but in the future it will also allow to pick `3.4.4`, `3.5.0`, `3.5.8`, `3.9.2`… as long as it's `3.x` with `x≥4` (so it won't pick an hypothetical future `4.0.0`, which is a good thing because such a major version bump would mean a breaking change)

Thank you for your feedback. I will be changing it to `~> 3.4`